src/ImageMetadata.jl

```diff
@@ -66,87 +76,102 @@ AxisArrays.HasAxes(A::ImageMetaAxis) = AxisArrays.HasAxes{true}()
-            @boundscheck checkbounds(img.data, I...)
-            @inbounds img.data[I...] = val
+            @boundscheck checkbounds(data(img), I...)
+            @inbounds data(img)[I...] = val
             val
         end
     end
 end
 
 @inline function Base.getindex(img::ImageMetaAxis, ax::Axis, I...)
-    result = img.data[ax, I...]
+    result = data(img)[ax, I...]
     maybe_wrap(img, result)
 end
 @inline function Base.getindex(img::ImageMetaAxis, i::Union{Integer,AbstractVector,Colon}, I...)
-    result = img.data[i, I...]
+    result = data(img)[i, I...]
     maybe_wrap(img, result)
 end
 maybe_wrap(img::ImageMeta{T}, result::T) where T = result
 maybe_wrap(img::ImageMeta{T}, result::AbstractArray{T}) where T = copyproperties(img, result)
 
 
 @inline function Base.setindex!(img::ImageMetaAxis, val, ax::Axis, I...)
-    setindex!(img.data, val, ax, I...)
+    setindex!(data(img), val, ax, I...)
 end
 @inline function Base.setindex!(img::ImageMetaAxis, val, i::Union{Integer,AbstractVector,Colon}, I...)
-    setindex!(img.data, val, i, I...)
+    setindex!(data(img), val, i, I...)
 end
 
-Base.view(img::ImageMeta, ax::Axis, I...) = shareproperties(img, view(img.data, ax, I...))
-Base.view(img::ImageMeta{T,N}, I::Vararg{ViewIndex,N}) where {T,N} = shareproperties(img, view(img.data, I...))
-Base.view(img::ImageMeta, i::ViewIndex) = shareproperties(img, view(img.data, i))
-Base.view(img::ImageMeta, I::Vararg{ViewIndex,N}) where {N} = shareproperties(img, view(img.data, I...))
+Base.view(img::ImageMeta, ax::Axis, I...) = shareproperties(img, view(data(img), ax, I...))
+Base.view(img::ImageMeta{T,N}, I::Vararg{ViewIndex,N}) where {T,N} = shareproperties(img, view(data(img), I...))
+Base.view(img::ImageMeta, i::ViewIndex) = shareproperties(img, view(data(img), i))
+Base.view(img::ImageMeta, I::Vararg{ViewIndex,N}) where {N} = shareproperties(img, view(data(img), I...))
+
+function Base.getproperty(img::ImageMeta, propname::Symbol)
+    # TODO remove these once deprecations are done
+    if propname === :data
+        Base.depwarn("img.data is deprecated use data(img) to extract data img.", :data)
+        return data(img)
+    elseif propname === :properties
+        Base.depwarn("img.properties is deprecated use properties(img) to extract data img.", :properties)
+        return properties(img)
+    else
+        return properties(img)[propname]
+    end
+end
 
-Base.getindex(img::ImageMeta, propname::AbstractString) = img.properties[propname]
+function Base.setproperty!(img::ImageMeta, propname::Symbol, X)
+    return setindex!(properties(img), X, propname )
+end
 
-Base.setindex!(img::ImageMeta, X, propname::AbstractString) = setindex!(img.properties, X, propname)
+Base.propertynames(img::ImageMeta) = (keys(properties(img))...,)
 
-Base.copy(img::ImageMeta) = ImageMeta(copy(img.data), deepcopy(img.properties))
+Base.copy(img::ImageMeta) = ImageMeta(copy(data(img)), deepcopy(properties(img)))
```

Comment:
```suggestion
Base.copy(img::ImageMeta) = ImageMeta(copy(data(img)), copy(properties(img)))
```
`deepcopy` is really slow and I now think `copy` should not "silently" switch to `deepcopy`. This relies on having `copy` methods defined for all relevant types, of course.

If you agree with this change, we need to search/replace all uses of `deepcopy` (I'm not marking them below).

Reply:
Attempting to replace `deepcopy` and found out tests failed because of this behavior.
```julia
julia> d[:prop] = [1,2,3]
3-element Array{Int64,1}:
 1
 2
 3

julia> d2 = copy(d)
Dict{Symbol,Any} with 1 entry:
  :prop => [1, 2, 3]

julia> d2[:prop][2] = -2
-2

julia> d
Dict{Symbol,Any} with 1 entry:
  :prop => [1, -2, 3]
```

Reply:
So make a depth-2 copy then...

```julia
julia> new_prop = Dict(zip(keys(prop), copy.(values(prop))))
Dict{Symbol,Array{Int64,1}} with 1 entry:
  :time => [1, 2, 3]

julia> new_prop[:time][2] = -2
-2

julia> new_prop
Dict{Symbol,Array{Int64,1}} with 1 entry:
  :time => [1, -2, 3]

julia> prop
Dict{Symbol,Array{Int64,1}} with 1 entry:
  :time => [1, 2, 3]
```

Reply:
Here's the best work around I could come up with
```julia
d = Dict{Symbol,Any}()
d[:prop] = [1,2,3]

dict_copy1(x) = deepcopy(x)

dict_copy2(x) = typeof(x)(pairs(x)...)

julia> @btime dict_copy1($d)
  242.260 ns (7 allocations: 1.06 KiB)
Dict{Symbol,Any} with 1 entry:
  :prop => [1, 2, 3]

julia> @btime dict_copy2($d)
  234.756 ns (6 allocations: 672 bytes)
Dict{Symbol,Any} with 1 entry:
  :prop => [1, 2, 3]

d2 = dict_copy2(d)
d2[:prop][2] = -2
```

The real problem with any work around is that we can't assume that the property field is a `Dict` and therefore even using `typeof(x)(pairs(x)...)` assumes that we have a certain interface for constructing dictionaries. Although it seems like a reasonable assumption that dictionaries can all be constructed this way, there's not an explicit interface for dictionaries  defined.

If the behavior we're seeing with `copy(::AbstractDict)` is undesirable this seems like an issue that should be addressed in base. Otherwise maybe we should have `copy` and `deepcopy` defined here.

Reply:
OK, sounds like this is more involved than I had hoped. Let's stick with `deepcopy` for now.